solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -1014,8 +1014,10 @@ private void openNewSearcherAndUpdateCommitPoint() throws IOException {
     // must get the latest solrCore object because the one we have might be closed because of a
     // reload
     // todo stop keeping solrCore around
-    SolrCore core = solrCore.getCoreContainer().getCore(solrCore.getName());
-    try {
+    try (SolrCore core = solrCore.getCoreContainer().getCore(solrCore.getName())) {
+      if (core == null) {
+        return; // core closed, presumably
+      }
```

Comment:
Wondering if this would be an alternative:

```
if (core != null) {
  ...
}
```

i.e. no `return;` statement and then the `replicationHandler.indexCommitPoint = commitPoint;` assignment could amount to `replicationHandler.indexCommitPoint = null;` which depending on the prior value is not the same as not assigning because of the `return`.

Or if there's a `return` then could `IndexCommit commitPoint;` scope move into the `try` block?

Reply:
Shrug; I'm not sure if it'd be good if we go from non-null to null.   Instead, I could see using the current core as a fallback.  But I haven't worked with this code whatsoever and maybe you have.

Reply:
Hmm, yeah, the consequences of going from non-null to null are quite tricky to analyse here. So yeah, minimally avoiding the NPE seems safer then.